rts/src/main/java/eta/runtime/io/MemoryManager.java

```diff
@@ -210,13 +211,14 @@ public static long loadStrings(String[] ss, String charset) throws UnsupportedEn
             int size = 0;
             for (int i = 0; i < ss.length; i++) {
                 byte[] ssBytes = ss[i].getBytes(charset);
-                size += ssBytes.length;
+                size += ssBytes.length + 1;
                 bytess[i] = ssBytes;
             }
             address         = allocateBuffer(size, false);
             ByteBuffer dest = getBoundedBuffer(address);
             for (int i = 0; i < ss.length; i++) {
                 dest.put(bytess[i]);
+                dest.put((byte) 0);
```

Comment:
In the multi-string case, I think you should increment the size by 1 *outside* the loop since we only need a single 0 byte at the end.

`loadStrings` handles the case where you have a literal string that is larger than 65,535 bytes (and hence can't be stored in the constant pool), and so you split it up into smaller pieces and assemble them at runtime into a giant string.

Reply:
oh, i did not understand its intention